src/vs/workbench/services/textfile/common/textFileEditorModelManager.ts

```diff
@@ -158,7 +158,7 @@ export class TextFileEditorModelManager extends Disposable implements ITextFileE
 
 			// Install state change listener
 			this.mapResourceToStateChangeListener.set(resource, model.onDidStateChange(state => {
-				const event = new TextFileModelChangeEvent(model, state);
+				const event = new TextFileModelChangeEvent(model!, state);
```

Comment:
@mjbvz how come this is needed when `model` is being created a few lines above? Sounds like a bug in TS to me we should maybe report?

Reply:
This is not a TS bug. Model uses a `let` assignment and because we are referencing `model` inside of a closure, it is possible that someone outside of the closure could delete `model` at some point after it was set a few lines above this line.

The better fix is to introduce a new `const` assignments for `model` but I was fairly conservative in changing the code. This is the same issue for most of my uses of null-check suppressions. I'll look through to see which ones can be reasonably be fixed by adding a new const variable

Reply:
@mjbvz I feel like TS could be smarter here and find out that no-one is writing to model in the block. I will follow up in their repo with a smaller sample.

Reply:
https://github.com/Microsoft/TypeScript/issues/30224